test/simple.test.ts

```diff
@@ -0,0 +1,171 @@
+        }
+    });
+
+    it('can get frames at random times', async() => {
+        // Arrange
+        const extractor = await SimpleExtractor.create({
+            inputFileOrUrl: 'https://storage.googleapis.com/lumen5-prod-images/countTo60.mp4',
+        });
+        const times_to_get = [
+            1, // forward to 30
+            0, // backwards to 15
+            1.5, // forward to 45
+            0.5, // backward to 15
+        ];
+
+        // Act and Assert
+        for (let i = 0; i < times_to_get.length; i++) {
+            const imageData = await extractor.getImageDataAtTime(times_to_get[i]);
+            if (!imageData) {
+                continue;
+            }
+            const canvas = createCanvas(imageData.width, imageData.height);
+            const ctx = canvas.getContext('2d');
+            ctx.putImageData(imageData, 0, 0);
+            expect(canvas.toBuffer('image/png')).toMatchImageSnapshot();
+        }
+    });
+
+    it('can get the first 10 frames', async() => {
+        // Arrange
+        const extractor = await SimpleExtractor.create({
+            inputFileOrUrl: 'https://storage.googleapis.com/lumen5-prod-images/countTo60.mp4',
+        });
+
+        // Act & assert
+        // ensure we render the 2nd frame properly - if we read the next packet we'll draw 3 instead of 2
+        for (let i = 0; i < 10; i++) {
+            const time = i / 30.0 + FRAME_SYNC_DELTA;
+            const imageData = await extractor.getImageDataAtTime(time);
+            if (!imageData) {
+                continue;
+            }
+            const canvas = createCanvas(imageData.width, imageData.height);
+            const ctx = canvas.getContext('2d');
+            ctx.putImageData(imageData, 0, 0);
+            expect(canvas.toBuffer('image/png')).toMatchImageSnapshot();
+        }
+    });
+
+    it('can get the middle 10 frames', async() => {
+        // Arrange
+        // This test is pretty slow because our countTo60 video only has 1 I-frame. We have to run through all packets
+        // to get the last ones.
+        const extractor = await SimpleExtractor.create({
+            inputFileOrUrl: 'https://storage.googleapis.com/lumen5-prod-images/countTo60.mp4',
+        });
+
+        // Act & assert
+        // ensure we render the last few frames properly - we have to flush the decoder to get the last few frames
+        for (let i = 20; i < 30; i++) {
+            const time = i / 30.0 + FRAME_SYNC_DELTA;
+            const imageData = await extractor.getImageDataAtTime(time);
+            if (!imageData) {
+                continue;
+            }
+            const canvas = createCanvas(imageData.width, imageData.height);
+            const ctx = canvas.getContext('2d');
+            ctx.putImageData(imageData, 0, 0);
+            expect(canvas.toBuffer('image/png')).toMatchImageSnapshot();
+        }
+    });
+
+    it('can get the last 10 frames', async() => {
```

Comment:
We should integrate all tests from the previous version

Except tests which used the previous PNG dumping.